### ISSUE TITLE
Fully mange stop/suspend/resume/terminate action in plugin

### ISSUE THREAD

**Issue by Tan-JiaLiang** [argoproj/argo-workflows#10476]
# Summary
Current, plugin can not catch the stop/suspend/resume/terminate actions. So plugin maybe submit a job to another cluster like spark/flink, when we stop workflow, plugin can not get the stop action, so maybe plugin stopped, spark/flink job still running. 

We should provide a way for plugin's action catch.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by alexgeorgousis** [comment]
> Current, plugin can not catch the stop/suspend/resume/terminate actions

@Tan-JiaLiang what "plugin" are you referring to here?

**Comment by Tan-JiaLiang** [comment]
@alexgeorgousis see https://argoproj.github.io/argo-workflows/plugins/

**Comment by mio4kon** [comment]
Yes, I think this is a very important feature. The current plug-in cannot listen to other states of the workflow, resulting in the workflow using the plug-in being completely out of control.

**Comment by jswxstw** [comment]
#12441 has fixed the workflow terminate and stop problem with argo plugin, the argo plugin will be stopped since the agent pod will be deleted.
However, the jobs started by the argo plugin will not be stopped. I think the argo plugin needs to record all jobs it has started, and stop them after it receives the TERM signal.

**Comment by shuangkun** [comment]
Have you implemented spark or flink plugin or have any progress?

**Comment by csh0101** [comment]
are there support volcano
